### PR TITLE
misc v0.7 fixes

### PR DESCRIPTION
**Pull request by ZacLN**
fixes Unicode references that have been moved + a couple of other deprecations

### CODE REVIEW

File: src/lexer.jl [hunk 1]
```diff
@@ -53,8 +53,10 @@ tokenize(x, ::Type{RawToken}) = Lexer(x, RawToken)
 tokenize(x) = Lexer(x, Token)
 
 # Iterator interface
-Base.iteratorsize(::Type{Lexer{IO_t,T}}) where {IO_t,T} = Base.SizeUnknown()
-Base.iteratoreltype(::Type{Lexer{IO_t,T}}) where {IO_t,T} = Base.HasEltype()
+Base.IteratorSize(::Type{Lexer{IO_t,T}}) where {IO_t,T} = Base.SizeUnknown()
+# Base.iteratorsize(::Type{Lexer{IO_t,T}}) where {IO_t,T} = Base.SizeUnknown()
```
Comment (KristofferC) [inline src/lexer.jl]: Remove these comments?

File: src/utilities.jl [hunk 2]
```diff
@@ -26,17 +26,28 @@ The JuliaParser.jl package is licensed under the MIT "Expat" License:
 > SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 =#
 
-import Base.UTF8proc
+import Base.Unicode
+
+utf8_trailing = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```
Comment (KristofferC) [inline src/utilities.jl]: These need const?

**Review by KristofferC — COMMENTED** [review]
Any benchmarks? Would be interesting to see.

File: src/utilities.jl [hunk 3]
```diff
@@ -26,17 +26,28 @@ The JuliaParser.jl package is licensed under the MIT "Expat" License:
...
+0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 5]
+
+const utf8_offset = [0x00000000
+                    0x00003080
+                    0x000e2080
+                    0x03c82080
+                    0xfa082080
+                    0x82082080]
+# const EOF_CHAR = convert(Char,typemax(UInt32))
+const EOF_CHAR = '\e'
```
Comment (ZacLN) [inline src/utilities.jl]: This is wrong..., the deleted code wouldn't run on v0.7

Comment (KristofferC) [reply src/utilities.jl]: Use `typemax(Char)`

Comment (ZacLN) [reply src/utilities.jl]: cheers, all sorted

Comment (ZacLN) [reply src/utilities.jl]: Actually, `typemax` doesn't work

**Comment by ZacLN** [comment]
Tests run in ~ 35% less time

**Comment by KristofferC** [comment]
Benchmark takes 135 ms on 0.7 vs 158 ms on 0.6

**Pull request merged**